mlflow/pytorch/_lightning_autolog.py

```diff
@@ -438,15 +357,27 @@ def on_train_batch_end(
         batch,
         batch_idx,
     ) -> None:
+        if self.trainer is None:
```

Comment:
I feel a bit odd to set the trainer at `on_train_batch_end` hook, if we don't want to add it to the constructor, can we set it at `on_train_start()` hook?

Reply:
oh, adding it in constructor also works, and it is better.